src/pcp/gadgets.rs

```diff
@@ -192,6 +192,176 @@ impl<F: FieldElement> Gadget<F> for PolyEval<F> {
     }
 }
 
+/// The gadget for the MeanVarUnsignedVector type. It is not designed for general use.
+///
+/// MeanVarUnsigned is parameterized by a positive integer `bits`. Its arity is `2*bits+1`:
+///
+///  * The first `bits` values are interpreted as a vector `r_vec`.
+///  * The next `bits` values are interpreted as a vector `x_vec`.
+///  * The last value is interpreted as a singleton `x`.
+///
+/// The gadget is designed to output `x^2` *as long as* `x_vec` is a vector comprised of 0s and 1s.
+/// It first computes `w[l] = p(x_vec[l]) * r_vec[l]` for each `0 <= l < bits`, where `p(x) =
+/// x(x-1)`. It then computes `w = w[0] + ... + w[bits-1]` and returns `w + x^2`.
+///
+/// If `vec_x[l] == 1` or `vec_x[l] == 0`, then `p(vec_x[l]) == 0`; otherwise, `p(vec_x[l]) != 0`.
+/// The validity circuit for MeanVarUnsignedVector sets `r_vec[l] = r^(l+1)`, where `r` is a
+/// uniform random field element. This is ensures that, if `p(vec_x[l]) != 0` for some `l`, then
+/// `w != 0` with high probability.
+pub struct MeanVarUnsigned<F: FieldElement> {
+    poly: [F; 3],
```

Comment:
```suggestion
    /// Polynomial used to check that values are in range
    poly: [F; 3],
```

Reply:
Added a comment that's a bit more precise.